clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -400,15 +404,17 @@ object GarbageCollector {
 
         val region = getRegion(args)
 
-        remove(configMapper,
-               storageNSForSdkClient,
-               gcAddressesLocation,
-               expiredAddresses,
-               markID,
-               region,
-               storageType,
-               schema
-              )
+        val removed = remove(configMapper,
+                             storageNSForSdkClient,
+                             gcAddressesLocation,
+                             expiredAddresses,
+                             markID,
+                             region,
+                             storageType,
+                             schema
+                            )
+        logRunId(runID, storageType, storageNSForSdkClient, hc, region)
+        removed
       } else {
         spark.createDataFrame(spark.sparkContext.emptyRDD[Row], schema)
```

Comment:
In this case we lose the runID, just because we had nothing to sweep _this time_.  Not sure that makes sense.

Reply:
This is not the case.
In this case `shouldSweep == false`, thus it was explicitly requested to not sweep (passed in the `do_sweep=false` flag).
Even if there was nothing to sweep (i.e. the commits file was empty), yet `shouldSweep == true`, the above `if` clause will handle it and the run ID will be logged.

Reply:
```scala
// TODO(ariels): Implement "depends-on" between "review code" and "learn to read".
```